### DISCUSSION TITLE
How do I pass an ObservableCollection<KeyValuePair<double,string>> to a PieChart?

### DISCUSSION THREAD

**Discussion by Dennis-Scheurer-zz** [Live-Charts/LiveCharts2#437] [Q&A] [answered]
Hello,

i am struggling with a way to fill my PieCharts with an  ObservableCollection<KeyValuePair<double,string>>. I couldn't find the right information in the sample folder.

This is my code:

` public ObservableCollection<KeyValuePair<double, string>> GetEmployeeGenders()
        {
            ObservableCollection<KeyValuePair<double, string>> EmployeeGenders = new();

            foreach (IGrouping<string, Employee> group in new ObservableCollection<IGrouping<string, Employee>>(Employees.GroupBy(x => x.Gender).ToList()))
            {
               EmployeeGenders.Add(new KeyValuePair<double,string>(group.Count(), group.Key));
            }
          
            return EmployeeGenders;
        }`

I tried this:

` GenderChart = calc.GetEmployeeGenders().AsLiveChartsPieSeries<KeyValuePair<double, string>>();`

but it didnt work. How can I solve my problem? I will also need to give each gender a specific color in my PieChart.

Thanks for the help!!

**Comment by vadimffe** [comment, marked as answer]
If you haven't solved your issue, here is an example how to populate PieChart using Linq:

```
      IEnumerable<ISeries> result = this.DataRecords
            .Where(w => w.TimeNow.Date == this.SelectedDate.Date)
            .GroupBy(g => g.Running)
            .Select(item => new PieSeries<double>
            {
              Values = new List<double> { item.Sum(x => x.SecondsElapsed) },
              Name = item.Key ? "On" : "Off",
              AnimationsSpeed = new TimeSpan(0, 0, 0),
            });

      this.ChartSeries = new ObservableCollection<ISeries>(result);
```

and

```
    private IEnumerable<ISeries> chartSeries;
    public IEnumerable<ISeries> ChartSeries
    {
      get => this.chartSeries;
      set
      {
        this.chartSeries = value;
        this.OnPropertyChanged();
      }
    }
```

This example returns chart with two sections. You can adjust it by removing "if else" statement in Name
